### PR TITLE
Add add-user request via input form

### PR DESCRIPTION
**Pull request by matmuher**
TODO:
+ Also need to check response code
+ move to axios:
TLDR: It can be useful for adding authentication

Benefits of axios:

1. Interceptors: Axios provides built-in support for request and response interceptors. These allow you to add custom logic to all requests and responses globally, which can be very useful for tasks like authentication, logging, or modifying requests/responses.

2. Automatic transforms: Axios automatically converts the response data to JavaScript objects, saving you from manually parsing JSON responses.

3. Request cancellation: Axios provides methods to cancel ongoing requests, which can be useful in scenarios where you need to stop a request before it completes.

4. Easy error handling: Axios makes it simple to catch and handle errors, providing both error responses and rejected promises.

With axios code would look like:

```
import React from "react";
import UserList from "./UserList";
import { useNavigate } from 'react-router-dom';
import axios from 'axios'; // Make sure to install axios if not already installed

function UsersRenderer() {
  const navigate = useNavigate();
  const [inputValue, setInputValue] = React.useState('');

  const onClickAddUser = async () => {
    if (!inputValue.trim()) {
      alert('Please enter a name');
      return;
    }

    try {
      const response = await axios.post('http://localhost:80/add-user', { "name": inputValue });
      console.log(response.data);
      alert('User added successfully');
      setInputValue('');
    } catch (error) {
      console.error('Error adding user:', error);
      alert('Failed to add user');
    }
  };

  return (
    <div>
      <div>
        <input
          type="text"
          name="firstName"
          value={inputValue}
          onChange={(e) => setInputValue(e.target.value)}
        />
        <button onClick={onClickAddUser}>
          Add user
        </button>
        <UserList />
      </div>
    </div>
  );
}

export default UsersRenderer;

```

### CODE REVIEW

**Comment by matmuher** [comment]
Example of interceptor: import React from "react";

```
import UserList from "./UserList";
import { useNavigate } from 'react-router-dom';
import axios from 'axios';

// Create an Axios instance with interceptors
const api = axios.create({
  baseURL: 'http://localhost:80'
});

api.interceptors.request.use(
  (config) => {
    // Add a token to the request header
    config.headers['Authorization'] = 'Bearer YOUR_TOKEN_HERE';
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

api.interceptors.response.use(
  (response) => {
    // Log the response
    console.log('Response:', response.data);
    return response;
  },
  (error) => {
    // Handle error
    console.error('Error:', error.message);
    return Promise.reject(error);
  }
);

function UsersRenderer() {
  const navigate = useNavigate();
  const [inputValue, setInputValue] = React.useState('');

  const onClickAddUser = async () => {
    if (!inputValue.trim()) {
      alert('Please enter a name');
      return;
    }

    try {
      const response = await api.post('/add-user', { name: inputValue });
      console.log(response.data);
      alert('User added successfully');
      setInputValue('');
    } catch (error) {
      console.error('Error adding user:', error);
      alert('Failed to add user');
    }
  };

  // Rest of your component code remains the same
}

export default UsersRenderer;

```
